parser/jsonparser/jsonparser_test.go

```diff
@@ -0,0 +1,44 @@
+package jsonparser
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestJsonParse(t *testing.T) {
+	parser := NewJsonParser()
+	line := `{"time_local":"2021-02-03T11:22:33+08:00","request_length":123,"request_method":"GET","request":"GET /order/2145 HTTP/1.1","body_bytes_sent":518,"status": 200,"request_time":0.544,"upstream_response_time":"0.543"}`
+
+	got, err := parser.ParseString(line)
+	if err != nil {
+		t.Error(err)
+	}
+	want := map[string]string{
+		"time_local":             "2021-02-03T11:22:33+08:00",
+		"request_time":           "0.544",
+		"request_length":         "123",
+		"upstream_response_time": "0.543",
+		"status":                 "200",
+		"body_bytes_sent":        "518",
+		"request":                "GET /order/2145 HTTP/1.1",
+		"request_method":         "GET",
+	}
+
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("JsonParser.Parse() = %v, want %v", got, want)
+	}
+}
+
+func BenchmarkParseJson(b *testing.B) {
```

Comment:
Also, :+1: for thinking of adding benchmarks!

Reply:
I did some experiments using other JSON parsing packages, such as easyjson. They can be faster, but they are more strict with the format. Maybe we can discuss this in a separate issue.

Reply:
If I remember correctly, easyjson basically needs statically typed structs, doesn't it? This would probably make supporting arbitrary log data structures... difficult, to say the least. However, that discussion should not block merging this change, so I think continuing the discussion in a separate issue (as suggested) might be a good idea.